test/OpenTelemetry.Instrumentation.AspNetCore.Tests/BasicTests.cs

```diff
@@ -348,7 +348,7 @@ public async Task ExtractContextIrrespectiveOfSamplingDecision(SamplingDecision
                 var expectedTraceId = ActivityTraceId.CreateRandom();
                 var expectedParentSpanId = ActivitySpanId.CreateRandom();
                 var expectedTraceState = "rojo=1,congo=2";
-                var activityContext = new ActivityContext(expectedTraceId, expectedParentSpanId, ActivityTraceFlags.Recorded, expectedTraceState);
+                var activityContext = new ActivityContext(expectedTraceId, expectedParentSpanId, ActivityTraceFlags.Recorded, expectedTraceState, true);
```

Comment:
Was setting `isRemote = true` required for this test?

Reply:
Yes - the extracted context should have `isRemote = true` in case of ASP.NET Core incoming request. This change is not completely related to .NET7.0. Even without that it should use `isRemote = true`. It relates to the check [here] (https://github.com/open-telemetry/opentelemetry-dotnet/blob/f471a9f197d797015123fe95d3e12b6abf8e1f5f/src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs#L94)

Reply:
Can you remind me of the significance of that check? 😆

Pretty sure I understood it at one point, but I'm staring at it and it's purpose is no longer obvious. Is the problem that IsRemote is not properly handled from a parent context provided when ASP.NET Core creates its activity?

Also, it looks like the test app does not return the value of IsRemote https://github.com/open-telemetry/opentelemetry-dotnet/blob/3a90287360604762db9ac077223fe3b45c266650/test/TestApp.AspNetCore.7.0/Controllers/ChildActivityController.cs#L34-L36. If we're setting it shouldn't we assert on it?

Reply:
Slight correction: I added `isRemote = true` due to following reason:

The  `isRemote = true` is needed as that impacts whether we create the activity or not in case when the sampler decides to drop the activity (we have that case in this test). The change is explained in #3315. So I think any tests we do where are comparing contexts of the extracted parent we should add this. (`isRemote` should always be true for parent context of activity created by ASP.NET Core).

**Explanation of check**: The check is used in case when a custom propagator is used by user e.g. B3. In this case the parent extracted by ASP.NET Core library will not have correct value as it only understands TraceContextPropagator. If that check determines the parent context extracted by custom propagator is different than the one extracted by library, it will create a new activity with correct parent context set. (In this case the sampler will re-run again with correct parent context passed in as one of the [parameters](https://github.com/open-telemetry/opentelemetry-dotnet/blob/f471a9f197d797015123fe95d3e12b6abf8e1f5f/src/OpenTelemetry/Trace/TracerProviderSdk.cs#L371). This is done so that the correct context is propagated to next hop.

Let me know if that makes sense :)


For the assert part, I don't think we can check that as we are creating a child activity of activity created by framework (which will have `HasRemoteParent` set as true). Not sure why we create new activity there and not do Activity.Current. Will have to check.

Reply:
> Let me know if that makes sense :)

Yes, thank you that's helpful for jogging my memory. Regarding the new Activity, I _think_ it might be because if ASP.NET Core also extracted a valid context that is different from the one we extract we don't want to clobber it. Though I don't really recall why that would be bad. In some scenarios it almost seems like it could be useful. Anyways... this is off topic. I might study this code more and consider adding a comment to remind my future self 😄.